flaskr.py

```diff
@@ -34,50 +35,105 @@ def init_db():
-	return redirect(url_for('show_entries'))
+    if not session.get('logged_in'):
+        abort(401)
+    db = get_db()
+    max_order_id = db.execute('SELECT MAX(sort_order) FROM entries').fetchone()[0]
+    if max_order_id is None:
+        new_order_id = 10
+    else:
+        new_order_id = max_order_id + 10
+    db.execute(
+        'insert into entries (title, text, sort_order) values (?, ?, ?)',
+        [
+            request.form['title'],
+            request.form['text'],
+            new_order_id
+        ]
+    )
+    db.commit()
+    flash('New entry was successfuly posted')
+    return redirect(url_for('show_entries'))
 
 @app.route('/login', methods=['GET','POST'])
 def login():
-	error=None
-	if request.method == 'POST':
-		if request.form['username'] != app.config['USERNAME']:
-			error = 'Invalid username'
-		elif request.form['password'] != app.config['PASSWORD']:
-			error = 'Invalid password'
-		else:
-			session['logged_in'] = True
-			flash('You were logged in')
-			return redirect(url_for('show_entries'))
-	return render_template('login.html', error=error)			
+    error=None
+    if request.method == 'POST':
+        if request.form['username'] != app.config['USERNAME']:
+            error = 'Invalid username'
+        elif request.form['password'] != app.config['PASSWORD']:
+            error = 'Invalid password'
+        else:
+            session['logged_in'] = True
+            flash('You were logged in')
+            return redirect(url_for('show_entries'))
+    return render_template('login.html', error=error)
+
 
 @app.route('/logout')
 def logout():
-	session.pop('logged_in', None)
-	flash('You were logged out')
-	return redirect(url_for('show_entries'))
\ No newline at end of file
+    """Log user out."""
+    session.pop('logged_in', None)
+    flash('You were logged out')
+    return redirect(url_for('show_entries'))
+
+
+@app.route('/entry=<entry_id>&move=<direction>', methods=['POST'])
+def move_entry(entry_id, direction):
+    """Move entry up or down."""
+    db = get_db()
+    moving_entry_so = db.execute(
+        'select sort_order from entries where id = (?)',
+        (int(entry_id),)
+    ).fetchone()[0]
+    if fnmatch.fnmatch(direction, 'up'):
+        another_entry_id_so = db.execute(
+            '''select id, sort_order from entries
+            where sort_order > (?)
+            order by sort_order asc limit 1''',
+            (moving_entry_so,)
+        ).fetchone()
+    elif fnmatch.fnmatch(direction, 'down'):
+        another_entry_id_so = db.execute(
+            '''select id, sort_order from entries
+            where sort_order < (?)
+            order by sort_order desc limit 1''',
+            (moving_entry_so,)
+        ).fetchone()
+    try:
+        temp_id = another_entry_id_so[0]
+        temp_so = another_entry_id_so[1]
+        db.execute(
+            'UPDATE entries set sort_order = ? where id = ?',
+            (moving_entry_so, temp_id)
+        )
+        db.commit()
```

Comment:
Yes, I agree, I was wondering do I need two db.commit() in the code, just forgot to ask :) thanks for the clarification